lib/ansible/modules/system/etc_hosts.py

```diff
@@ -0,0 +1,429 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils._text import to_bytes
+import os
+import re
+import socket
+import tempfile
+
+
+ANSIBLE_METADATA = {
+    'status': [
+        'preview'
+    ],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: etc_hosts
+author: Jiri Tyr (@jtyr)
+version_added: '2.3'
+short_description: Manages records in the /etc/hosts file
+description:
+  - Add, remove and modify records in the /etc/hosts file.
+options:
+  alias:
+    required: false
+    default: null
+    description:
+      - Alias(es) of the IP address. Multiple aliases can be either a normal
+        YAML list or a string containing comma separated list.
+  hostname:
+    required: false
+    description:
+      - Canonical hostname of the IP address. Required if I(state=present).
+  ip:
+    required: true
+    description:
+      - IP address.
+  path:
+    required: false
+    default: /etc/hosts
+    description:
+      - Path to the hosts file. Required if I(state=present).
+  state:
+    required: false
+    choices: [absent, present]
+    default: present
+    description:
+      - State of the record in the hosts file.
+      - If C(present), I(ip) and I(hostname) are required.
+      - If C(absent), I(ip) or I(hostname) is required.
+
+extends_documentation_fragment:
+  - files
+'''
+
+EXAMPLES = '''
+- name: Add a new record without alias
+  etc_hosts:
+    ip: 10.0.0.1
+    hostname: some.domain1.com
+
+- name: Add a new record with hostname for the same IP
+  etc_hosts:
+    ip: 10.0.0.1
+    hostname: some.domain2.com
+
+- name: Add a new record with another IP for the same hostname
+  etc_hosts:
+    ip: 10.0.0.2
+    hostname: some.domain3.com
+
+- name: Add a new record with hostname and alias for the same IP
+  etc_hosts:
+    ip: 10.0.0.3
+    hostname: some.domain3.com
+    alias: some
+
+- name: Add a new record with multiple aliases
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    alias:
+      - some
+      - other
+
+- name: Update the list of aliases for the existing IP and hostname
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    alias:
+      - some
+      - other
+      - alias
+
+- name: Remove all records with the specified IP
+  etc_hosts:
+    ip: 10.0.0.1
+    state: absent
+
+- name: Remove all records with the specified hostname
+  etc_hosts:
+    hostname: some.domain3.com
+    state: absent
+
+- name: Remove record with specified IP and hostname
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    state: absent
+'''
+
+RETURN = '''
+state:
+  description: state of the target, after execution
+  returned: success
+  type: string
+  sample: "present"
+'''
+
+
+class EtcHosts:
```

Comment:
@abadger I can fix that if that's really so important. But there is another 33 modules using exactly the same syntax and they don't seem to break anything:

```
$ egrep -R '^\s*class .[^(]*:' *
cloud/docker/_docker.py:class ContainerSet:
cloud/centurylink/clc_modify_server.py:class ClcModifyServer:
cloud/centurylink/clc_blueprint_package.py:class ClcBlueprintPackage:
cloud/centurylink/clc_firewall_policy.py:class ClcFirewallPolicy:
cloud/centurylink/clc_alert_policy.py:class ClcAlertPolicy:
cloud/centurylink/clc_aa_policy.py:class ClcAntiAffinityPolicy:
cloud/centurylink/clc_server.py:class ClcServer:
cloud/centurylink/clc_loadbalancer.py:class ClcLoadBalancer:
cloud/centurylink/clc_server_snapshot.py:class ClcSnapshot:
cloud/amazon/ec2_elb.py:class ElbManager:
cloud/amazon/ecs_ecr.py:class EcsEcr:
cloud/amazon/ecs_cluster.py:class EcsClusterManager:
cloud/amazon/rds.py:class RDSConnection:
cloud/amazon/rds.py:class RDS2Connection:
cloud/amazon/rds.py:class RDSDBInstance:
cloud/amazon/rds.py:class RDS2DBInstance:
cloud/amazon/rds.py:class RDSSnapshot:
cloud/amazon/rds.py:class RDS2Snapshot:
cloud/amazon/cloudtrail.py:class CloudTrailManager:
cloud/amazon/cloudfront_facts.py:class CloudFrontServiceManager:
cloud/amazon/cloudformation_facts.py:class CloudFormationServiceManager:
cloud/amazon/ecs_service_facts.py:class EcsServiceManager:
cloud/amazon/lambda_event.py:class AWSConnection:
cloud/amazon/ecs_taskdefinition.py:class EcsTaskManager:
cloud/amazon/lambda_alias.py:class AWSConnection:
cloud/amazon/ecs_task.py:class EcsExecManager:
cloud/amazon/ec2_customer_gateway.py:class Ec2CustomerGatewayManager:
cloud/amazon/ecs_service.py:class EcsServiceManager:
cloud/google/gce.py:class LazyDiskImage:
cloud/misc/xenserver_facts.py:class XenServerFacts:
clustering/consul_acl.py:class Rules:
clustering/consul_acl.py:class Rule:
monitoring/icinga2_feature.py:class Icinga2FeatureHelper:
network/dnsmadeeasy.py:class DME2:
network/omapi_host.py:class OmapiHostManager:
packaging/os/rpm_key.py:class RpmKey:
packaging/language/maven_artifact.py:class MavenDownloader:
remote_management/stacki/stacki_host.py:class StackiHost:
web_infrastructure/jenkins_job.py:class JenkinsJob:
```